contract/src/pegasus.js

```diff
@@ -271,11 +187,23 @@ const makeCourier = ({
     const localAmount = packetToLocalAmount(packet);
     const depositFacet = await E(board).getValue(depositAddress);
 
+    const { userSeat, zcfSeat } = zcf.makeEmptySeatKit();
+
     // Redeem the backing payment.
-    const payment = await redeem(localAmount);
+    try {
+      redeem(zcfSeat, { Transfer: localAmount });
+      zcfSeat.exit();
+    } catch (e) {
+      zcfSeat.kickOut(e);
+      throw e;
+    }
 
-    // Send to the deposit facet, if we can.
-    return E(depositFacet).receive(payment);
+    const payout = await E(userSeat).getPayout('Transfer');
+
+    // Send the payout promise to the deposit facet.
+    E(depositFacet)
```

Comment:
`Receiver` (the type of the `receive` function) returns a `Promise<void>`, presumably so that it doesn't settle until the job is done. The old code returned the promise for the result of the eventual send. The new code has no `return`, causing the returned promise to settle after the `receive` message is sent but possibly before it is received.

Reply:
We don't want to wait for the `receive` message to return so that the deposit facet can't hang up (i.e. DoS) an ordered channel.  Once we've gotten to this point, their payment is committed and won't be refunded on a failed `receive`.

Reply:
Good! Please make that apparent when reading the code, so others don't jump to the same misimpression that I did. A final commented `return undefined;` would do it.

Reply:
Okay, great.  I've done that.